tests/test_channelfile.py

```diff
@@ -6,23 +6,29 @@
 class ChannelFileBase(object):
     @patch("paramiko.channel.ChannelFile._set_mode")
     def test_defaults_to_unbuffered_reading(self, setmode):
-        self.klass(Channel(None))
+        self.klass(Channel(None), encoding="utf8", errors="strict")
```

Comment:
You've added `encoding="utf8"` cases to many tests, but I don't see any tests that probe compatibility with other encodings. Those would be good to add.

Reply:
Right. In the first place this is supposed to "not break the current tests".

Adding one or two different test cases where the system does not use utf8 would be very welcome. I'll try to put something together, but I'm not yet sure at how to do it.

We have to keep in mind that there are two different problems: Encoding of file content and encoding of file paths including system messages. (Welcome message on login and error messages etc)

Reply:
Ah, I see -- with this PR, you're going for a ~zero-effect change, making sure that adding the `encoding` and `errors` arguments didn't break anything in the default `utf8` use case? Thus, setting the stage for future development/testing when new encodings enter the mix?

Reply:
Yes. That's right 🙂